vignettes/big_ns.Rmd

```diff
@@ -0,0 +1,176 @@
+---
+title: "Big N's"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Big N's}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup, message = FALSE, warning = FALSE}
+library(tfrmt)
+library(tibble)
+library(dplyr)
+```
+
+Adding "Big N's" aka subject totals to columns. Is a very common and nessicary part of building tables for cinical trials. {tfrmt} allows you to do this by using a `big_n_structure` to specify which values in the ARD should be added to the columns and how they should be formatted. 
+
+
+To dynamically add big N's to column labels first the information needs to be in the ARD. For a simple example let's take a small demographics table that just has Age ad Sex for Placebo, Treatment and Total. 
+```{r}
+
+data <- tibble(Group = rep(c("Age (y)", "Sex", "Age (y)", "Sex"), c(3, 3, 6,12)),
+               Label = rep(c("n", "Mean (SD)", "Male","Female"), c(6, 6,6,6)),
+               Column = rep(c("Placebo", "Treatment", "Total"), times = 8),
+               Param = rep(c("n", "mean", "sd", "n", "pct", "n", "pct"),  c(6, 3, 3, 3,3,3,3)),
+               Value = c(15,13,28,14,13,27,73.56, 74.231,71.84,9.347,7.234,8.293,8,7,15,8/14,7/13,15/27,6,6,12,6/14,6/13,12/27
+               )
+) %>%
+  # Note because tfrmt only does rounding we will need to have the percents multiplied by 100
+  mutate(Value = case_when(Param == "pct" ~ Value * 100,
+                           TRUE ~ Value),
+         ord1 = if_else(Group == "Age (y)", 1, 2),
+         ord2 = if_else(Label == "n", 1, 2))
+data
+```
+
+
+In this study we have 30 subjects per Arm, for a total of 60 subjects. To make the N's go to the right place we need to ensure the values in the `Column` column match the rest of the data. Then we can combine this information with teh data above to make the full ARD for this table.  
```

Comment:
```suggestion
In this study we have 30 subjects per treatment arm for a total of 60 subjects. Because the ARD is 1 row per unique value, our Ns will contribute 3 rows to the ARD. To make the N's go to the right place, we need to supply information about the `column` column values. The `column` column values should map to those already available in the ARD. Then we can combine this information with the data above to make the full ARD for this table.  
```